### PR TITLE
field3d: import from homebrew/science

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Companion PR: Homebrew/homebrew-science#6599
Cross reference: Homebrew/homebrew-science#6331

### CODE REVIEW

File: Formula/field3d.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+class Field3d < Formula
+  desc "Library for storing voxel data on disk and in memory"
+  homepage "https://sites.google.com/site/field3d/"
+  url "https://github.com/imageworks/Field3D/archive/v1.7.2.tar.gz"
+  sha256 "8f7c33ecb4489ed626455cf3998d911a079b4f137f86814d9c37c5765bf4b020"
+
+  depends_on "scons" => :build
+  depends_on "boost"
+  depends_on "ilmbase"
+  depends_on "hdf5"
```
Comment (ilovezfs) [inline Formula/field3d.rb]: alphabetize non-build deps amongst themselves

File: Formula/field3d.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  def install
+    scons
+    include.install Dir["install/**/**/release/include/*"]
```
Comment (ilovezfs) [inline Formula/field3d.rb]: why aren't we using `scons "install"`?

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/field3d.rb (hunk 2 above)
Comment (EricFromCanada) [reply Formula/field3d.rb]: According to the [README](https://sites.google.com/site/field3d/building-field3d#TOC-Building-Field3D-on-Mac-OS-X), when using `scons` for building it just places the built files in an "install" directory, relying on a package manager (MacPorts in their example) to do the installing.

File: Formula/field3d.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    man1.install "man/f3dinfo.1"
+    pkgshare.install "contrib", "test", "apps/sample_code"
+  end
+
+  test do
+    system ENV.cxx, "-I#{include}", "-L#{lib}", "-lField3D",
+           "-I#{Formula["boost"].opt_include}",
+           "-L#{Formula["boost"].opt_lib}", "-lboost_system",
+           "-I#{Formula["hdf5"].opt_include}",
+           "-L#{Formula["hdf5"].opt_lib}", "-lhdf5",
```
Comment (ilovezfs) [inline Formula/field3d.rb]: also needs `"-I#{Formula["ilmbase"].opt_include}",`

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks @EricFromCanada! :rocket:

**Pull request closed without merging**